File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.XtraEditors;

namespace LookUpListWithNullEntry {
	public partial class Form1 : Form {
		public Form1() {
			InitializeComponent();

            FillData();

            this.repositoryItemLookUpEdit1.DataSource = new DataSourceWithEmptyItem(this.xpc2);
		}

        private void FillData() {
            if(xpc2.Count == 0) {
                Lookup lookup = new Lookup();
                lookup.Description = "Description 1";
                lookup.Save();
                xpc2.Add(lookup);
            }
        }

        private void repositoryItemLookUpEdit1_ButtonClick(object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e) {
            if(e.Button.Kind == DevExpress.XtraEditors.Controls.ButtonPredefines.Delete) {
                LookUpEdit editor = (LookUpEdit)sender;
                editor.CancelPopup();
                editor.EditValue = null;
            }
        }
	}

    public class Main : XPObject {
        public string Name;
        public Lookup Description;
    }

    public class Lookup : XPObject {
        public string Description;
    }
}

File: VB/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports DevExpress.XtraEditors

Namespace LookUpListWithNullEntry

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            FillData()
            repositoryItemLookUpEdit1.DataSource = New DataSourceWithEmptyItem(xpc2)
        End Sub

        Private Sub FillData()
            If xpc2.Count = 0 Then
                Dim lookup As Lookup = New Lookup()
                lookup.Description = "Description 1"
                lookup.Save()
                xpc2.Add(lookup)
            End If
        End Sub

        Private Sub repositoryItemLookUpEdit1_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs)
            If e.Button.Kind = DevExpress.XtraEditors.Controls.ButtonPredefines.Delete Then
                Dim editor As LookUpEdit = CType(sender, LookUpEdit)
                editor.CancelPopup()
                editor.EditValue = Nothing
            End If
        End Sub
    End Class

    Public Class Main
        Inherits XPObject

        Public Name As String

        Public Description As Lookup
    End Class

    Public Class Lookup
        Inherits XPObject

        Public Description As String
    End Class
End Namespace
